api/v1beta1/ovs_types.go

```diff
@@ -0,0 +1,111 @@
+/*
+Copyright 2022.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package v1beta1
+
+import (
+	"github.com/openstack-k8s-operators/lib-common/modules/common/condition"
+
+	corev1 "k8s.io/api/core/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+)
+
+// OVSSpec defines the desired state of OVS
+type OVSSpec struct {
+	// +kubebuilder:validation:Optional
+	// +kubebuilder:default=1
+	// +kubebuilder:validation:Maximum=1
+	// +kubebuilder:validation:Minimum=0
+	// Replicas of OVS to run
+	Replicas int32 `json:"replicas,omitempty"`
+
+	// +kubebuilder:validation:Optional
+	ExternalIDS OVSExternalIDs `json:"external-ids"`
+
+	OvsContainerImage string `json:"ovsContainerImage,omitempty"`
+
+	OvnContainerImage string `json:"ovnContainerImage,omitempty"`
+
+	// +kubebuilder:validation:Optional
+	NicMappings map[string]string `json:"nic_mappings,omitempty"`
+
+	// +kubebuilder:validation:Optional
+	// Resources - Compute Resources required by this service (Limits/Requests).
+	// https://kubernetes.io/docs/concepts/configuration/manage-resources-containers/
+	Resources corev1.ResourceRequirements `json:"resources,omitempty"`
+
+	// +kubebuilder:validation:Optional
+	// NodeSelector to target subset of worker nodes running this service
+	NodeSelector map[string]string `json:"nodeSelector,omitempty"`
```

Comment:
For cases where we don't specifiy Nodeselector it will launch on all openshift nodes, may be we want to limit it only to very few nodes?

Reply:
but do we know to which ones?

Reply:
right, I don't think there is a good default, or you set a default here and add a required step in the docs to label the nodes you want to get ovs deployed with this selector, or chose a custom one if modified.